docs/userguide/networking/dockernetworks.md

```diff
@@ -0,0 +1,368 @@
+
+### A bridge network
+
+The easiest user-defined network to create is a `bridge` network. This network is similar to the historical, default `docker0` network. There are some added features and some old features that aren't available.
+
+```
+$ docker network create -d bridge isolated_nw
+8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7
+
+$ docker network inspect isolated_nw
+{
+    "name": "isolated_nw",
+    "id": "8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7",
+    "driver": "bridge",
+    "containers": {}
+}
+
+$ docker network ls
+NETWORK ID          NAME                DRIVER
+9f904ee27bf5        none                null
+cf03ee007fb4        host                host
+7fca4eb8c647        bridge              bridge
+8b05faa32aeb        isolated_nw         bridge
+
+```
+
+After you create the network, you can launch containers on it using  the `docker run --net=<NETWORK>` option.
+
+```
+$ docker run --net=isolated_nw -itd --name=container3 busybox
+777344ef4943d34827a3504a802bf15db69327d7abe4af28a05084ca7406f843
+
+$ docker network inspect isolated_nw
+{
+    "name": "isolated_nw",
+    "id": "8b05faa32aeb43215f67678084a9c51afbdffe64cd91e3f5bb8267475f8bf1a7",
+    "driver": "bridge",
+    "containers": {
+        "777344ef4943d34827a3504a802bf15db69327d7abe4af28a05084ca7406f843": {
+            "endpoint": "c7f22f8da07fb8ecc687d08377cfcdb80b4dd8624c2a8208b1a4268985e38683",
+            "mac_address": "02:42:ac:14:00:01",
+            "ipv4_address": "172.20.0.1/16",
+            "ipv6_address": ""
+        }
+    }
+}
+```
+
+The containers you launch into this network must reside on the same Docker host.  Each container in the network can immediately communicate with other containers in the network. Though, the network itself isolates the containers from external networks.
+
+![](images/bridge_network.png)
+
+Within of a user-defined bridge network, linking is not supported. You can expose and publish container ports on containers in this network. This is useful if you want make a portion of the `bridge` network available to an outside network.
+
+![](images/network_access.png)
+
+A bridge network is useful in cases where you want to run a relatively small network on a single host. You can, however, create significantly larger networks by creating an `overlay` network.
+
+
+### An overlay network
+
+Docker's `overlay` network driver supports multi-host networking natively
+out-of-the-box. This support is accomplished with the help of `libnetwork`, a
+built-in VXLAN-based overlay network driver, and Docker's `libkv` library.
+
+The `overlay` network requires a valid key-value store service. Currently, Docker's supports Consul, Etcd, Zookeeper (Distributed store) and
+BoltDB (Local store). Before creating a network you must install and configure your chosen key-value store service.  The Docker hosts that you intend to network and the service must be able to communicate.
```

Comment:
couple of comments here.
1. I dont think we should mention BoltDB here. it is an internal implementation for local network objects and not exposed/configurable by the user.
2. This is networking section and I don't think adding information such as TLS configuration for KV Store here is appropriate. But, I do agree that we might need a independent section just for the KV Store configuration.